renderdoc/driver/d3d12/d3d12_initstate.cpp

```diff
@@ -79,6 +79,10 @@ bool D3D12ResourceManager::Prepare_InitialState(ID3D12DeviceChild *res)
         SetInitialContents(GetResID(r), D3D12InitialContents(D3D12InitialContents::MapDirect));
         return true;
       }
+      else if(heapProps.Type == D3D12_HEAP_TYPE_UPLOAD)
```

Comment:
Can you explain this change? Unless I'm misunderstanding this seems like it would mean there are no initial states for `UPLOAD` heap resources.

Reply:
Yes, you are right, thank you. The problem was that you cannot transition UPLOAD resources into COPY_SOURCE state for CopyResource. But it appears that you don't need to, it just works.

As a note, ReadFromSubresource() appeared to be invalid for reading from buffers and CopyResource() (which is described as GPU side copying) surprisingly works.

Reply:
Great, thanks. I think UPLOAD resources should always be forced into COPY_SOURCE state so it might work out anyway? Either way having an extra check for safety does no harm.